### PR TITLE
RGB sliders behaving like CSP/Krita, Hex editbox and various bugfixes by justburner (with modifications)

### PR DESCRIPTION
**Pull request by shun-iwasawa**
This PR is based on #4222 by @justburner , with some modifications as follows:

- Enabled to make the color sliders to be conventional one by choosing option from hamburger menu in the Style Editor.
    styleeditor2
    
    - The appearance setting will be stored in the user's .env file and is shared by all Style Editor panels.
    - Default value is `Relative colored + Triangle handle` (i.e. the new one)
- The style preview chips are now with double height. When the style with color parameters (sub colors) is selected, the color parameter selector becomes visible and the style preview chips shrink to the conventional height.
  styleeditor4

-  Related to the above changes, the MyPaint style chip in the Palette Viewer is modified so that the style icon is shown without stretching.
   styleeditor5

### CODE REVIEW

**Comment by shun-iwasawa** [comment]
@RodneyBaker 
Please do not use `Squash and merge` when merging this PR to make the commits separated in the history. Thanks!

**Comment by RodneyBaker** [comment]
LGTM!

WIll give @justburner a chance to wake up and double check to see if anything is missing.

**Comment by RodneyBaker** [comment]
If someone could double check drawing with standard raster levels.
I seem to be having an issue where no matter what color on the palette I use I get a dark almost black color.
I've check to make sure I don't have any of the Ink and Paint checks toggled on.

**Comment by justburner** [comment]
> 

I see you picked the Tahoma2D approach by moving the icons to the left. lol
Overall all the changes looks good to me. 👍 

> 

This is one of those settings the user pick own preference once and forget about it, so, really didn't expected this approach. OwO
At least it makes it easier to find it so that's good. 

> 

Nice approach, i like it. 😄
Anyway I'm surprised you didn't stretched the "Auto" & "Apply" buttons vertically similar to how they were in OpenToonz 1.0

> 

This was a oversight from my PR, really cool addition. 💯

**Comment by gab3d** [comment]
@shun-iwasawa  nice work!

- Could there be a little less space destined to the slider arrows? especially in lower res screens they seem unnecessarily big.
- Perhaps if HEX characters could share space with the Auto Apply buttons, the panel could be functional while being narrower than with the current layout. (it also could have the benefit of having all the values aligned in the panel).
- if the two lower right buttons get stacked, we gain space for having the chips bigger also.

imagen
imagen

**Comment by konero** [comment]
@shun-iwasawa I'm getting a crash by doing the following:
1. Go to the 'Drawing' room
2. Try resize the Style Editor to the smallest it allows you to go by dragging the dock handle

If I let go of the mouse button quickly I can avoid a crash, but if I keep the mouse button held as if making precise adjustment I can reproduce the crash over and over.

Otherwise, this is nicely done! The sliders feel much nicer to use now.

**Comment by shun-iwasawa** [comment]
@gab3d 

> Could there be a little less space destined to the slider arrows? especially in lower res screens they seem unnecessarily big.

At least for the first request you can customize the button size by applying the additional style sheet like as follows:

```
#colorSliderAddButton,
#colorSliderSubButton {
  width: 14;
  padding: -2;
}
```
image

**Comment by shun-iwasawa** [comment]
@konero Thank you for reporting the crash, I'll check it tomorrow!

File: toonz/sources/toonzqt/styleeditor.cpp [hunk 1]
```diff
@@ -1020,79 +1258,68 @@ void ColorSlider::paintEvent(QPaintEvent *event) {
...
+    w -= 3;
+  } else {
+    x += s_chandle_size / 2;
+    w -= s_chandle_size;
+    h -= 3;
+    if (isLineHandle) {
+      y += 1;
+      h -= 2;
+    }
+  }
```
Comment (justburner) [inline toonz/sources/toonzqt/styleeditor.cpp:1277]: Seems i should had tested both orientations, not just the default one alone.
This is one way to solve the crash:
```suggestion
  if (w < 2) w = 2;
  if (h < 2) h = 2;
  
```
it seems `makeLinearShading` doesn't like when size is 1 or lower.

Comment (shun-iwasawa) [reply toonz/sources/toonzqt/styleeditor.cpp:1277]: Confirmed that the suggested change resolves the crash. Thank you @justburner !

**Comment by RodneyBaker** [comment]
@shun-iwasawa @justburner  Just doing final checks but all is looking good.
Thank you for the update.
Also thanks to @konero and @gab3d for testing and feedback.

LGTM

**Pull request merged**